src/Events/BaseEvent.php

```diff
@@ -26,29 +27,43 @@ public function __serialize(): array
                 'name' => $this->getWorkflowName(),
             ],
         ];
+        if (!$this instanceof GuardEvent) {
+            $result['context'] = $this->getContext();
+        }
```

Comment:
Rather than the awkward type-checking here, I think this should be done on the child class instead.

Reply:
I think there is a trait in the base class (`HasContextTrait`) could you simply check the base class is using that trait and if it is, then attach the context?